sample.ts

```diff
@@ -0,0 +1,52 @@
+import fc from "fast-check";
+
+/**
+ * Script function to compare two fast-check generators by sampling them.
+ * @param generatorA The first generator to compare.
+ * @param generatorB The second generator to compare.
+ * @param numSamples The number of samples to compare.
+ * @returns An array of result objects containing the sample number, values
+ * generated by each generator, and the match object containing whether the
+ * values match and the mismatched items.
+ */
+export const compareGenerators = (
+  generatorA: fc.Arbitrary<any>,
+  generatorB: fc.Arbitrary<any>,
+  numSamples: number
+) => {
+  const results = [];
+  for (let i = 0; i < numSamples; i++) {
+    const seed = 123 * i;
+
+    const valueA = fc.sample(generatorA, { seed: seed });
+    const valueB = fc.sample(generatorB, { seed: seed });
```

Comment:
Why we even need `compareGenerators` and `getMismatchedItems`. I'd delete the `sample.ts` file.

---

What I'd do instead in the actual tests we have is just

```ts
const valueA = fc.sample(generatorA, { seed: seed });
const valueB = fc.sample(generatorB, { seed: seed });
```

and compare the two values in the simplest possible way, e.g. JSON.stringify.

No need for loops:
- Just one run of fc.sample on both generators is enough.
- If the test is [erratic](https://wubw.github.io/2017/xunit_test_patterns/#erratic-test) we'll catch it sooner or later running the test-suite (as part of the CI or our everyday workflow).